Scripts/Editor/ThrusterEditor.cs

```diff
@@ -0,0 +1,191 @@
+            EditorGUILayout.FloatField("Time since force request", myThruster.timeSinceForceRequest);
+            EditorGUI.EndDisabledGroup();
+
+            ThrusterSO.ApplyModifiedProperties();
+        }
+
+        private List<string> getThursterNames()
+        {
+            List<string> nameList = new List<string>();
+            foreach(var element in myThruster.thrusters)
+            {
+                nameList.Add(element.name);
+            }
+            return nameList;
+        }
+
+        /// <summary>
+        /// Getting all instances of ThrusterAsset in the project
+        /// </summary>
+        public static T[] GetAllInstances<T>() where T : ScriptableObject
+        {
+            string[] guids = AssetDatabase.FindAssets("t:"+ typeof(T).Name);
+            T[] a = new T[guids.Length];
+            for(int i =0;i<guids.Length;i++)
+            {
+                string path = AssetDatabase.GUIDToAssetPath(guids[i]);
+                a[i] = AssetDatabase.LoadAssetAtPath<T>(path);
+            }
+            return a;
+        }
+
+        /// <summary>
+        /// Saving new thruster if current by new name. Adding (n)
+        /// to the name of the thruster if current name allready exists
+        /// </summary>
+        public void SaveNewThruster(string name, AnimationCurve curve)
+        {
+            ThrusterAsset newThruster = new ThrusterAsset();
+            newThruster.name = name;
+            CopyCurve(newThruster.curve, curve);
+            string path = thrusterFolderPath + name + ".asset";
+            AssetDatabase.CreateAsset(newThruster, path);
+            myThruster.thrusters = GetAllInstances<ThrusterAsset>();
+            myThruster.selectedThrusterIndex =  myThruster.thrusters.ToList().FindIndex((x)=>x.name == name);
+            myThruster.previousThrusterIndex =-1;
+        }
+
+        public bool AreCurvesDifferent(AnimationCurve a1, AnimationCurve a2)
+        {
+            if (a1.length != a2.length) return true;
+            for (int i = 0; i<a2.length; i++)
+            {
+                if(a1.keys[i].value != a2.keys[i].value || a1.keys[i].time != a2.keys[i].time)
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        public string getInitialSavingName(string thrusterName)
+        {
+            int i = 1;
+            while (true)
+            {
+                if(!myThruster.thrusters.Where(x => x.name.Contains(thrusterName + "(" + i + ")")).Any())
+                {
+                    break;
+                }
+                else
+                {
+                    i++;
+                }
+            }
+            string returnName = thrusterName + "(" + i + ")";
+            return returnName;
+        }
+
+        public void CopyCurve(AnimationCurve a1, AnimationCurve a2)
+        {
+            a1.keys = a2.keys;
```

Comment:
this is ok. Try using AnimationCurve constructor with param arguments. Maybe it creates a deepcopy